.github/workflows/security.yml

```diff
@@ -0,0 +1,46 @@
+name: Security Check
+
+on:
+  push:
+    branches:
+      - main
+
+  pull_request:
+    branches:
+      - "*"
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+jobs:
+  security-audit:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout source
+        uses: actions/checkout@8e5e7e5ab8b370d6c329ec480221332ada57f0ab # v3.5.2
+
+      - uses: gradle/wrapper-validation-action@v1
+
+      - uses: actions/setup-java@v3
+        with:
+          distribution: "corretto"
+          java-version: 11
+
+      - uses: actions/cache@v3
+        with:
+          path: ~/.gradle/caches
+          key: gradle-${{ runner.os }}-${{ hashFiles('**/*.gradle.kts') }}-${{ hashFiles('**/gradle/wrapper/gradle-wrapper.properties') }}-${{ hashFiles('**/versions.properties') }}
+
+      - name: Install Snyk
+        uses: snyk/actions/setup@3e2680e8df93a24b52d119b1305fb7cedc60ceae # latest master (no released tag)
+
+      - name: Snyk VULN and License Check Test
+        run: snyk test --all-projects --sarif-file-output=snyk.sarif
+        env:
+          SNYK_TOKEN: ${{ secrets.SNYK_TOKEN }}
+
+      - name: Upload SARIF result to GitHub Code Scanning
+        uses: github/codeql-action/upload-sarif@74483a38d39275f33fcff5f35b679b5ca4a26a99 # v2.22.5
```

Comment:
Curious why pin this to a commit?

Reply:
We pin commits purposefully because we try to get deterministic builds (for a given commit sha, that triggered this build branch, we want the same results, though not always perfect)... More details in the TBD Security Supply Chain doc [here](https://docs.google.com/document/d/1r22jO93WO44SNHawyrFhNhvT04G1ay9m3uf37wi15H8/edit?pli=1#heading=h.fypobv6byrbg).

Reply:
The build will still be reproducible if you user `@master` or a tagged commit. If you fix that commit, then any vulnerability found in the build tool (i.e. in the upload-sarif) will not be updated. How would you mitigate that?

Reply:
This is a good call. @ALRubinger and I discussed about this point yesterday, and what we actually want is to make sure that for a given Git commit sha in our source code that our generated artifacts (binaries, jars etc) will have the same sha too. 

To achieve this I was restricting beyond what was needed by pinning the versions of everything in the pipeline, whereas in reality only the versions of dependencies used in our SDK should be pinned.

This is to avoid not only inconsistent artifacts sha but also introducing unexpected behavior in our sdks for our clients,  introducing new vulns etc.

But yes, you are right, we will probably be benefitted when the supporting tools in our pipeline are upgraded!

[TBD Supply Chain Security doc](https://docs.google.com/document/d/1r22jO93WO44SNHawyrFhNhvT04G1ay9m3uf37wi15H8/edit#heading=h.u50axc8wrgxq) (check the Reproducible Builds and Locked Versions sections)

Reply:
Yes this case is a choice of "choose your poison":

* Pin the dependency: always get reliable results from the build
* Let the dependency upgrade: `upload-sarif` may update over time or work differently and break builds

**In this case I think `@master` is the right call**. It doesn't impact the binary artifact generated, may receive important updates over time, and the scanning done here relies on an external service that doesn't have a pinned version anyway. Props to both @andresuribe87 for catching and @leordev in protecting against version ranges.